### PR TITLE
💻 Re-structuring front-end data flow

### PR DESCRIPTION
**Pull request by lesliexin**
## Purpose
Pulling out the base context provider into its own file.

## Learning
https://www.netlify.com/blog/2020/12/01/using-react-context-for-state-management-in-next.js/

### CODE REVIEW

File: client/utils/EventsUtils.tsx [hunk 1]
```diff
@@ -0,0 +1,15 @@
+import { Id, EventInfo } from "../context/Base";
+
+export const formatEventsData = (eventsData) => {
+    const eventsMap = new Map<Id, EventInfo>();
+    eventsData.forEach(event => {
```
Comment (jeffzh4ng) [inline client/utils/EventsUtils.tsx:5]: same comment here

File: client/context/Base.tsx [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+}
+
+export interface EventInfo {
+    name: string
+    description: string
+}
+
+export type Id = number;
+
+export const clubsData = [
```
Comment (jeffzh4ng) [inline client/context/Base.tsx:21]: Alternative name

```suggestion
export const clubs = [
```

File: client/context/Base.tsx [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+        description: 'chef curry with the shot',
+    },
+    {
+        id: 111,
+        name: 'Loo Labs',
+        description: '👩‍🔬',
+    }
+];
+
+export const eventsData = [
```
Comment (jeffzh4ng) [inline client/context/Base.tsx:39]:
```suggestion
export const events = [
```

**Review by jeffzh4ng — CHANGES_REQUESTED** [review]

File: client/pages/events/[id].tsx [hunk 4]
```diff
@@ -1,16 +1,15 @@
...
-import { AppContext } from "../_app";
+import { useAppContext } from "../../context";
 
 export default function EventDetail() {
-  const appData = useContext(AppContext);
+  const { events } = useAppContext();
   const router = useRouter();
 
   const { id } = router.query;
   if (typeof id !== 'string') return null;
```
Comment (ShreyasPrasad) [inline client/pages/events/[id].tsx:10]: we'll probably make a nice 404 for this later :)

**Review by ShreyasPrasad — COMMENTED** [review]
looks good, agree with the small changes requested by Jeff.

File: client/context/Base.tsx (hunk 2 above)
Comment (lesliexin) [reply client/context/Base.tsx:21]: the state variables are already called `clubs` and `events`. we can rename to `CLUBS` and `EVENTS` instead?

**Review by jeffzh4ng — DISMISSED** [review]

**Comment by jeffzh4ng** [comment]
LGTM 🏆

**Review by ShreyasPrasad — APPROVED** [review]
looks good!

**Review by jeffzh4ng — APPROVED** [review]

**Review by ShreyasPrasad — APPROVED** [review]

**Pull request merged**